integreat_cms/cms/forms/organizations/organization_form.py

```diff
@@ -26,3 +34,48 @@ class Meta:
         widgets = {
             "icon": IconWidget(),
         }
+
+    def __init__(self, **kwargs):
+        r"""
+        Initialize organization form
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+        """
+        super().__init__(**kwargs)
+        if self.instance:
+            # logger.info(Organization.users.all())
+            logger.info(self.instance.slug)
+            logger.info(self.instance.pages.all())
+            # logger.info(self.instance.users.all())
+        self.fields["slug"].required = False
+
+    def clean_slug(self):
+        """
+        Validate the slug field (see :ref:`overriding-modelform-clean-method`)
+
+        :return: A unique slug based on the input value
+        :rtype: str
+        """
+        return generate_unique_slug_helper(self, "organization")
+
+    def clean(self):
+        """
+        Validate if form fields name is not already in use for another organization in the same region, see :meth:`django.forms.Form.clean`
+
+        :return: The cleaned form data
+        :rtype: dict
+        """
+        cleaned_data = super().clean()
+
+        if Organization.objects.filter(name=cleaned_data["name"]).exists():
```

Comment:
> I added a check for a already existing name in the organization forms clean method.

Thanks! Three small points:
1. If you validate only one field, I'd prefer the `clean_<fieldname>` over the general `clean()` method.
2. If you don't exclude the current organization instance, you can only create organizations and not save them (to e.g. change the icon), because the currently edited organization blocks its own name...
3. You did not limit the queryset to the current region, so it checks whether the value is in use globally instead of only region-wide.


```suggestion
    def clean_name(self):
        """
        Validate if form fields name is not already in use for another organization in the same region
        (see :ref:`overriding-modelform-clean-method`)

        :return: The name which is unique per region
        :rtype: str
        """
        cleaned_name = self.cleaned_data["name"]
        if (
            Organization.objects.exclude(id=self.instance.id)
            .filter(region=self.instance.region, name=cleaned_name)
            .exists()
        ):
```

> We might want to check later if we also want to not allow similar names such as names that only differ in a lower case / upper case character

Yeah, I think for now it's ok, no need for a more sophisticated check.

Reply:
Adjusted :+1: